### PR TITLE
[dynamo][sourceless] Return UserDefinedObjectVariable for unhandled objects

### PR DESCRIPTION
**Pull request by anijain2305**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #131414
* #131405
* #131389
* #131378
* #131367
* #131347

Fixes https://github.com/pytorch/pytorch/issues/130533

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

### CODE REVIEW

**Review by williamwen42 — APPROVED** [review]

**Review by mlazos — CHANGES_REQUESTED** [review]
I don't think this is a good idea; it will hide issues with SourcelessBuilder and mislead a developer into thinking it's working as intended. Curious if there was a specific use case that drove this? I think I would support this if we already had parity tests with the standard VariableBuilder but we don't today.

**Comment by anijain2305** [comment]
@mlazos The usecase is TensorDict and I also remember seeing other issues with similar graph breaks. I agree with your assessment that sourceless builder is not at parity with variable builder, and also that it can hide issues (like using a more narrow variable tracker). 

At the same time, I also think that UserDefinedObjectVariable is not so bad. For `fullgraph=True`, any missing opportunity will be quickly exposed. For `fullgraph=False`, the situation is not so different from today, where once someone works on graph breaks, the opportunity will soon be visible. 

I think I need to maybe make the warning loud or quickly visible, but I don't know where to put it.

File: torch/_dynamo/variables/builder.py [hunk 1]
```diff
@@ -2632,6 +2632,8 @@ def create(tx, value) -> VariableTracker:
             return DeviceMeshVariable(value)
         elif isinstance(value, re.Pattern):
             return RegexPatternVariable(value)
+        elif inspect.isclass(type(value)):
```
Comment (anijain2305) [inline torch/_dynamo/variables/builder.py:2635]: Bleh .. this is always True

**Review by jansel — CHANGES_REQUESTED** [review]
Without a source, how will we property guard on things?

**Pull request closed without merging**